pipenv/patched/piptools/utils.py

```diff
@@ -21,16 +22,20 @@
 def clean_requires_python(candidates):
     """Get a cleaned list of all the candidates with valid specifiers in the `requires_python` attributes."""
     all_candidates = []
+    py_version = parse_version(os.environ.get('PIP_PYTHON_VERSION', str(sys.version_info[:3])))
```

Comment:
`parse_version(str(sys.version_info[:3])` is not going to work(at least from my testing). Should use `'.'.join(map(str, sys.version_info[:3]))`